pcsx2/MemoryCardFolder.cpp

```diff
@@ -2063,7 +2063,15 @@ void FileAccessHelper::WriteIndex(const std::string& baseFolderName, MemoryCardF
 	pxAssert(entry->IsFile());
 
 	std::string folderName(baseFolderName);
-	parent->GetPath(&folderName);
+	if (parent != nullptr)
+	{
+		parent->GetPath(&folderName);
+	}
+	else
```

Comment:
Is baseFolderName always a non-empty string? If the else path is taken then no assignment is made to folderName, so folderName will copy the value from baseFolderName and never be written again. Fine as long as baseFolderName is guaranteed to be (correctly) populated, problematic if its possible that it too could be empty.

Reply:
It doesn't seem like it can be empty, baseFolderName is gotten from m_folderName which is set when the card is opened.

https://github.com/PCSX2/pcsx2/blob/7d1e7f2b58ea839563310ae1b284ac15c780f17e/pcsx2/MemoryCardFolder.cpp#L168-L173

and there is only one call to FileAccessHelper::WriteIndex
https://github.com/PCSX2/pcsx2/blob/7d1e7f2b58ea839563310ae1b284ac15c780f17e/pcsx2/MemoryCardFolder.cpp#L1369-L1372